### PR TITLE
EZP-29846: ezxmltext -> ricktext conversion: Conversion tool skips some ezxmltext rows

### PR DESCRIPTION
**Pull request by vidarl**
This one fixes [EZP-29846](https://jira.ez.no/browse/EZP-29846), also reported here : https://github.com/ezsystems/ezplatform-xmltext-fieldtype/issues/60

So the problem was that the different rows was distributed to child processes using `select * from ezcontentobject_attribute where data_type_string='ezxmltext' limit $limit, $offset`, which ended badly as data_type_string values where changed from 'ezxmltext' to 'ezrichtext' during conversion.
This PR has the quick fix for it. A better quicker solution would be to get ridd of the slower `$offset` approach all together

### CODE REVIEW

File: lib/FieldType/XmlText/Persistence/Legacy/ContentModelGateway.php [hunk 1]
```diff
@@ -92,18 +92,40 @@ public function getContentTypeFieldTypeUpdateQuery($fromFieldTypeIdentifier, $to
...
+            } else {
+                $query->orWhere(
+                    $query->expr()->eq(
+                        'a.data_type_string',
+                        ':datatypestring' . $index
+                    )
+                );
+            }
+            $query->setParameter(':datatypestring' . $index, $datatypeName);
+            ++$index;
```
Comment (andrerom) [inline lib/FieldType/XmlText/Persistence/Legacy/ContentModelGateway.php]: You don't need to do it like this, you can simply do a $query->expr()->in().
Like here: https://github.com/ezsystems/ezpublish-kernel/pull/2480/files#diff-01c5542d76ccdcabdbce4f0c3d00f06cR98

But using `PARAM_STR_ARRAY `

Same below

Comment (vidarl) [reply lib/FieldType/XmlText/Persistence/Legacy/ContentModelGateway.php]: Thank you. I actually had the impression that in() is slower than or. But this seems to no longer be the case, at least in mysql.

Fixed

**Comment by andrerom** [comment]
Should this target 1.7?

**Comment by vidarl** [comment]
FYI : fixed in

**Pull request closed without merging**